File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/cheats/ui/CheatItem.java
// SPDX-License-Identifier: GPL-2.0-or-later

package org.dolphinemu.dolphinemu.features.cheats.ui;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import org.dolphinemu.dolphinemu.features.cheats.model.Cheat;

public class CheatItem
{
  public static final int TYPE_CHEAT = 0;
  public static final int TYPE_HEADER = 1;
  public static final int TYPE_ACTION = 2;

  private final @Nullable Cheat mCheat;
  private final int mString;
  private final int mType;

  public CheatItem(@NonNull Cheat cheat)
  {
    mCheat = cheat;
    mString = 0;
    mType = TYPE_CHEAT;
  }

  public CheatItem(int type, int string)
  {
    mCheat = null;
    mString = string;
    mType = type;
  }

  @Nullable
  public Cheat getCheat()
  {
    return mCheat;
  }

  public int getString()
  {
    return mString;
  }

  public int getType()
  {
    return mType;
  }
}

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/cheats/ui/CheatItem.kt
// SPDX-License-Identifier: GPL-2.0-or-later

package org.dolphinemu.dolphinemu.features.cheats.ui

import org.dolphinemu.dolphinemu.features.cheats.model.Cheat

class CheatItem {
    val cheat: Cheat?
    val string: Int
    val type: Int

    constructor(cheat: Cheat) {
        this.cheat = cheat
        string = 0
        type = TYPE_CHEAT
    }

    constructor(type: Int, string: Int) {
        cheat = null
        this.string = string
        this.type = type
    }

    companion object {
        const val TYPE_HEADER = 0
        const val TYPE_CHEAT = 1
        const val TYPE_ACTION = 2
    }
}
